posthog/queries/session_recordings/session_recording.py

```diff
@@ -94,10 +100,81 @@ def get_metadata(self) -> RecordingMetadata:
 
         first_snapshot = all_snapshots[0]
 
-        return RecordingMetadata(
+        recording_metadata = RecordingMetadata(
             start_time=first_event_timestamp,
             end_time=last_event_timestamp,
             duration=last_event_timestamp - first_event_timestamp,
             session_id=first_snapshot.session_id,
             distinct_id=first_snapshot.distinct_id,
         )
+        if include_active_segments:
+            recording_metadata["active_segments_by_window_id"] = self._get_active_segments_by_window_id(all_snapshots)
+
+        return recording_metadata
+
+    def _is_active_event(self, event: SnapshotData) -> bool:
```

Comment:
Nit: would this suffice as a static method?

Reply:
Good call - I pulled it out to the session_recording helpers